apps/site/util/objects.ts

```diff
@@ -1,4 +1,21 @@
-export default function deepMerge<Obj1 extends object, Obj2 extends object>(
+type DebounceFunction<T = unknown> = (...args: Array<T>) => void;
+
+let timeoutId: NodeJS.Timeout;
+
+export const debounce =
+  <T extends DebounceFunction>(
+    func: T,
+    delay: number
+  ): ((...args: Parameters<T>) => void) =>
+  (...args: Parameters<T>) => {
+    clearTimeout(timeoutId);
+
+    timeoutId = setTimeout(() => {
+      func(...args);
```

Comment:
This can lead to loss of context from the original caller of the debouncing function